lib/kino_maplibre/map_cell.ex

```diff
@@ -115,6 +124,18 @@ defmodule KinoMapLibre.MapCell do
     end
   end
 
+  defp build_arg_source(nil, _), do: nil
+  defp build_arg_source(_, nil), do: nil
+  defp build_arg_source(id, data), do: [id, [type: :geojson, data: data]]
+
+  defp apply_node(%{args: nil}, acc), do: acc
+
+  defp apply_node(%{field: _field, name: function, module: module, args: args}, acc) do
+    quote do
+      unquote(acc) |> unquote(module).unquote(function)(unquote_splicing(args))
+    end
```

Comment:
Could you just use [`apply/3`](https://hexdocs.pm/elixir/Kernel.html#apply/3) here?

```suggestion
    apply(module, function, [acc | args])
```

The concat in the args is because you want your `acc` as first argument, right?

Reply:
Nervermind. You are generating the quoted code :)